// @shopify/cli/dist/cli/commands/demo/index.js
import {
  demo
} from "../../../chunk-OPMRCCAR.js";
import "../../../chunk-WBWFZ6IT.js";
import {
  base_command_default
} from "../../../chunk-WF3T42FP.js";
import "../../../chunk-46VGUYHL.js";
import "../../../chunk-6MJ52OLD.js";
import "../../../chunk-W7UHREAK.js";
import "../../../chunk-7SNGGY5B.js";
import "../../../chunk-5FVTOE7H.js";
import "../../../chunk-VU576BU3.js";
import "../../../chunk-474I4TNL.js";
import "../../../chunk-RJ2SJHQV.js";
import {
  readFile
} from "../../../chunk-6VZG567Q.js";
import "../../../chunk-SA32EMWV.js";
import "../../../chunk-JCAIZOYO.js";
import "../../../chunk-QI5VKSMW.js";
import {
  require_lib
} from "../../../chunk-3MMF34G4.js";
import "../../../chunk-YLPWACJJ.js";
import "../../../chunk-SMULJJG3.js";
import "../../../chunk-LOF3ZXQM.js";
import "../../../chunk-F6GZ5SSL.js";
import "../../../chunk-WI3E2P3B.js";
import "../../../chunk-CYG4QPQ3.js";
import "../../../chunk-P52UWCZD.js";
import {
  cwd,
  resolvePath
} from "../../../chunk-M336AND6.js";
import "../../../chunk-AXXAG7WU.js";
import "../../../chunk-566AK3QG.js";
import {
  __toESM,
  init_cjs_shims
} from "../../../chunk-BVGYIZ3O.js";

// src/cli/commands/demo/index.ts
init_cjs_shims();
var import_core = __toESM(require_lib(), 1);
var Demo = class _Demo extends base_command_default {
  static {
    this.description = "Demo a command design defined in a file";
  }
  static {
    this.hidden = true;
  }
  static {
    this.flags = {
      file: import_core.Flags.string({
        hidden: false,
        description: "The path to the design file.",
        env: "SHOPIFY_FLAG_PATH",
        parse: async (input) => resolvePath(input),
        default: async () => cwd()
      })
    };
  }
  async run() {
    const { flags } = await this.parse(_Demo);
    const contents = await readFile(flags.file);
    const design = JSON.parse(contents);
    await demo(design);
  }
};
export {
  Demo as default
};
